packages/plugin-types/API.md

```diff
@@ -74,10 +74,10 @@ export type BackendIdentifier = {
 };
 
 // @public (undocumented)
-export type BackendOutput = Record<string, BackendOutputEntry>;
+export type BackendOutput<T = Record<string, string>> = Record<string, BackendOutputEntry<T>>;

```

Comment:
why is this needed?

Reply:
Because `buckets` is `Record<string, string>[]`, but I can't set it in the type since it breaks other categories.

Parameter `T` was introduced so that we can restrict it only for Storage.

Reply:
I think by adapting the concept suggested here https://github.com/aws-amplify/amplify-backend/pull/1742#discussion_r1681550455 we could nest buckets under single storage output and avoid this.

Reply:
The doesn't affect here because the buckets array will be under single storage, which is not acceptable for `BackendOutput` before the current change with `T`. I saw the following error:

```
Type '{ name: string; bucket_name: string; aws_region: string; }[]' is not assignable to type 'string'.
```

Reply:
Perhaps we should do something like this https://github.com/aws-amplify/amplify-backend/blob/7361ff4ed9da10f9b48e327c96eff25eb81a9e10/packages/backend-output-schemas/src/function/v1.ts#L6 .

Reply:
- [ ] need to figure out why API.md doesn't update properly